v2/test/pre-release/pre_release_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package multitenant_test
+
+import (
+	"testing"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1beta20201101"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	resourceGroupName     = "aso-sample-rg"
+	preReleaseNamespace   = "pre-release"
+	vnetBeforeUpgradeName = "vnet-before-upgrade"
+)
+
+func Test_Pre_Release_ResourceCanBeCreated_BeforeUpgrade(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	tc.Namespace = preReleaseNamespace
+
+	rg := tc.NewTestResourceGroup()
+	rg.Name = resourceGroupName
+
+	vnet := newVnet(tc, vnetBeforeUpgradeName)
+	// We want to use 'CreateResourceUntracked' here as we don't want the resource cleanup for this test. We will be checking the
+	// backward compatibility for these resources after we upgrade the controller.
+	tc.CreateResourceAndWaitWithoutCleanup(rg)
+	tc.CreateResourceAndWaitWithoutCleanup(vnet)
+
+}
+
+func newVnet(tc *testcommon.KubePerTestContext, name string) *network.VirtualNetwork {
+	vnet := &network.VirtualNetwork{
+		ObjectMeta: tc.MakeObjectMetaWithName(name),
+		Spec: network.VirtualNetworks_Spec{
+			Owner: &genruntime.KnownResourceReference{
+				Name: resourceGroupName,
+			},
+			Location: tc.AzureRegion,
+			AddressSpace: &network.AddressSpace{
+				AddressPrefixes: []string{"10.0.0.0/8"},
+			},
+		},
+	}
+	return vnet
+}
+
+func Test_Pre_Release_ResourceCanBeCreated_AfterUpgrade(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	tc.Namespace = preReleaseNamespace
+
+	// TODO: Will have to change the version here when we go from beta to stable
+	vnetAfterUpgrade := newVnet(tc, tc.Namer.GenerateName("vn"))
+	tc.CreateResourceAndWait(vnetAfterUpgrade)
+	tc.DeleteResourcesAndWait(vnetAfterUpgrade)
+
+	// Patch for existing resource. This resource already will exist in kind as will be created without cleanup in test 'Test_Pre_Release_ResourceCanBeCreated_BeforeUpgrade'.
```

Comment:
How do you control the sequence of execution?

I thought `go test` was deliberately non-deterministic in the order it executes tests.

Reply:
Yep, I decided to use these tests instead of `kubectl` so that they can fail/pass when something goes wrong. We're controlling them through the task [here](https://github.com/Azure/azure-service-operator/pull/2483/files#diff-cd2d359855d0301ce190f1ec3b4c572ea690c83747f6df61c9340720e3d2425eR264)